sentry-android-core/src/main/java/io/sentry/android/core/AppLifecycleIntegration.java

```diff
@@ -110,23 +110,27 @@ private void addObserver(final @NotNull IHub hub) {
   }
 
   private void removeObserver() {
-    ProcessLifecycleOwner.get().getLifecycle().removeObserver(watcher);
+    final @Nullable LifecycleWatcher watcherRef = watcher;
```

Comment:
Is the `watcherRef` necessary? An `if` on the `watcher` should be enough, shouldn't it?

Reply:
Yes, as watcher could be set to `null` on another thread